azurelinuxagent/ga/remoteaccess.py

```diff
@@ -0,0 +1,158 @@
+            if self.os_util is None:
+                self.os_util = get_osutil()
+            if self.os_util.jit_enabled:
+                self.protocol = self.protocol_util.get_protocol()
+                current_incarnation = self.protocol.get_incarnation()
+                if self.incarnation != current_incarnation:
+                    # something changed. Handle remote access if any.
+                    self.incarnation = current_incarnation
+                    self.protocol.client.update_goal_state(True)
+                    self.protocol.client.update_remote_access_conf(self.protocol.client.goal_state)
+                    self.remote_access = self.protocol.client.remote_access
+                    if self.remote_access is not None:
+                        self.handle_remote_access()
+        except Exception as e:
+            msg = u"Exception processing remote access handler: {0}".format(
+                ustr(e))
+            logger.error(msg)
+            add_event(AGENT_NAME,
+                      version=CURRENT_VERSION,
+                      op=WALAEventOperation.HandleRemoteAccess,
+                      is_success=False,
+                      message=msg)
+            return
+
+    def handle_remote_access(self):
+        logger.verbose("Entered handle_remote_access")
+        if self.remote_access is not None:
+            # Get JIT user accounts.
+            all_users = self.os_util.get_users()
+            jit_users = set()
+            for usr in all_users:
+                if usr[4] == REMOTE_ACCESS_ACCOUNT_COMMENT:
+                    jit_users.add(usr[0])
+            for acc in self.remote_access.user_list.users:
+                raw_expiration = acc.expiration
+                account_expiration = datetime.strptime(raw_expiration, REMOTE_USR_EXPIRATION_FORMAT)
+                now = datetime.utcnow()
+                if acc.name not in jit_users and now < account_expiration:
+                    self.add_user(acc.name, acc.encrypted_password, account_expiration)
+
+    def add_user(self, username, encrypted_password, account_expiration):
+        created = False
+        try:
+            expiration_date = (account_expiration + timedelta(days=1)).strftime(DATE_FORMAT)
+            logger.verbose("[RemoteAccessHandler::add_user]: Adding user {0} with expiration date {1}".format(username, expiration_date))
+            self.os_util.useradd(username, expiration_date, REMOTE_ACCESS_ACCOUNT_COMMENT)
+            created = True
+            prv_key = os.path.join(conf.get_lib_dir(), TRANSPORT_PRIVATE_CERT)
+            pwd = self.cryptUtil.decrypt_secret(encrypted_password, prv_key)
+            self.os_util.chpasswd(username, pwd, conf.get_password_cryptid(), conf.get_password_crypt_salt_len())
+            self.os_util.conf_sudoer(username)
+            logger.info("[RemoteAccessHandler::add_user]: User '{0}' added successfully with expiration in {1}".format(username, expiration_date))
+            return
```

Comment:
Unnecessary returns, and below

Reply:
This one I want, it's the one below that's not needed.

---
In reply to: [194026975](https://github.com/Azure/WALinuxAgent/pull/1171#discussion_r194026975) [](ancestors = 194026975)